lib/govuk_publishing_components/presenters/select.rb

```diff
@@ -0,0 +1,42 @@
+module GovukPublishingComponents
+  module Presenters
+    class SelectHelper
+      attr_reader :options, :option_markup, :selected_option
+
+      def initialize(local_assigns)
+        @options = local_assigns[:options] if local_assigns[:options]
+        @option_markup = get_options
+      end
+
+    private
+
+      def get_options
+        return {} if @options.nil?
+
+        formatted_options = []
+
+        options.each do |option|
+          formatted_options << [
+            option[:text],
+            option[:value],
+            options_data_attribute(option[:data_attributes])
+          ]
+          @selected_option = option[:value] if option[:selected]
+        end
+
+        formatted_options
```

Comment:
If `options` is an array as [in the spec](https://github.com/alphagov/govuk_publishing_components/pull/650/files#diff-4d49ce1d3dc073aa8f98fb53fd8368dcR111), then we can use `map` and have a second method for the `find`. It's two operations but a little clearer.


```
def get_options
    return [] unless options.present?

    options.map { |option| {
        [ option[:text], option[:value], options_data_attribute(option[:data_attributes]) ]
    }
end

def selected_option
    return unless options.present?

    @selected_option ||= options.find { |option| option[:selected] }
end
```

Reply:
My Ruby knowledge isn't up to knowing which is better, but I'm slightly hesitant to do the loop twice if we don't need to. Let's catch up about this.

Reply:
In that case, you could add `@selected_option = option if option[:selected]` as the first line in the map block rather than using find.

Reply:
Updated, thanks.